pkg/services/alerting/result_handler.go

```diff
@@ -49,11 +49,12 @@ func (handler *defaultResultHandler) handle(evalContext *EvalContext) error {
 		handler.log.Info("New state change", "ruleId", evalContext.Rule.ID, "newState", evalContext.Rule.State, "prev state", evalContext.PrevAlertState)
 
 		cmd := &models.SetAlertStateCommand{
-			AlertId:  evalContext.Rule.ID,
-			OrgId:    evalContext.Rule.OrgID,
-			State:    evalContext.Rule.State,
-			Error:    executionError,
-			EvalData: annotationData,
+			AlertId:    evalContext.Rule.ID,
+			OrgId:      evalContext.Rule.OrgID,
+			State:      evalContext.Rule.State,
+			Error:      executionError,
+			EvalData:   annotationData,
+			NoDataFlag: evalContext.ToggleNoDataFlag(),
```

Comment:
Think that maybe the `evalContext.ToggleNoDataFlag()` should update some field in engine.go similar to how `GetNewState` is handed. What do you think?

Reply:
I agree, that's better. I will move it there.

Reply:
Well, the reason I originally put it there was that the alert is saved in the database before checking for sending a notification.
Therefore when the alert recovers from no_data (no_data -> pending -> ok), the NoDataFlag should be unset but `ShouldNotify` should see its previous value.
So, I think it cannot be moved to the `engine.go`.

Reply:
> should be unset but ShouldNotify should see its previous value

Hmm I see. That makes it a bit harder to reason about. Guess that's why we have a previous and a current state.